src/UIKit/UICellAccessory.cs

```diff
@@ -0,0 +1,114 @@
+//
+// UICellAccessory.cs
+//
+// Authors:
+//	Alex Soto <alexsoto@microsoft.com>
+//
+// Copyright (c) Microsoft Corporation.
+//
+
+using System;
+using System.Runtime.InteropServices;
+using Foundation;
+using ObjCRuntime;
+
+#nullable enable
+#if !WATCH
+
+namespace UIKit {
+
+	public partial class UICellAccessory {
+
+		[Watch (7,0), TV (14,0), iOS (14,0)]
+		[DllImport (Constants.UIKitLibrary)]
+		static extern IntPtr UICellAccessoryPositionBeforeAccessoryOfClass (IntPtr accessoryCls);
+
+		[Watch (7,0), TV (14,0), iOS (14,0)]
+		[return: DelegateProxy (typeof (SDUICellAccessoryPosition))]
+		[BindingImpl (BindingImplOptions.Optimizable)]
+		public static UICellAccessoryPosition GetPositionBeforeAccessory (Class accessoryClass)
+		{
+			if (accessoryClass == null)
+				throw new ArgumentNullException ("accessoryClass");
+			var ret = UICellAccessoryPositionBeforeAccessoryOfClass (accessoryClass.Handle);
+			return NIDUICellAccessoryPosition.Create (ret)!;
+		}
+		
+		[Watch (7,0), TV (14,0), iOS (14,0)]
+		[return: DelegateProxy (typeof (SDUICellAccessoryPosition))]
+		[BindingImpl (BindingImplOptions.Optimizable)]
+		public static UICellAccessoryPosition GetPositionBeforeAccessory (Type accessoryType) => GetPositionBeforeAccessory (new Class (accessoryType));
+
+		[Watch (7,0), TV (14,0), iOS (14,0)]
+		[DllImport (Constants.UIKitLibrary)]
+		static extern IntPtr UICellAccessoryPositionAfterAccessoryOfClass (IntPtr accessoryCls);
+
+		[Watch (7,0), TV (14,0), iOS (14,0)]
+		[return: DelegateProxy (typeof (SDUICellAccessoryPosition))]
+		[BindingImpl (BindingImplOptions.Optimizable)]
+		public static UICellAccessoryPosition GetPositionAfterAccessory (Class accessoryClass)
+		{
+			if (accessoryClass == null)
+				throw new ArgumentNullException ("accessoryClass");
+			var ret = UICellAccessoryPositionAfterAccessoryOfClass (accessoryClass.Handle);
+			return NIDUICellAccessoryPosition.Create (ret)!;
+		}
+
+		[Watch (7,0), TV (14,0), iOS (14,0)]
+		[return: DelegateProxy (typeof (SDUICellAccessoryPosition))]
+		[BindingImpl (BindingImplOptions.Optimizable)]
+		public static UICellAccessoryPosition GetPositionAfterAccessory (Type accessoryType) => GetPositionAfterAccessory (new Class (accessoryType));
+	} /* class UICellAccessory */
+
+	[UnmanagedFunctionPointerAttribute (CallingConvention.Cdecl)]
+	[UserDelegateType (typeof (UICellAccessoryPosition))]
+	internal delegate nuint DUICellAccessoryPosition (IntPtr block, IntPtr accessories);
+	
+	//
+	// This class bridges native block invocations that call into C#
```

Comment:
I wonder how complicated it would be to implement support for generating the supporting code for blocks by adding support in the generator for a `[GenerateBlockCode]` attribute to delegates:

```csharp
[GenerateBlockCode]
delegate nuint UICellAccessoryPosition (UICellAccessory [] accessories);
```